src/core/qgsgpkgflusher.cpp

```diff
@@ -90,8 +114,26 @@ void QgsGpkgFlusher::onLayersAdded( const QList<QgsMapLayer *> &layers )
   }
 }
 
+void QgsGpkgFlusher::stop()
+{
+  mFlusher->stop();
+}
+
+void QgsGpkgFlusher::start()
+{
+  mFlusher->start();
+}
+
+bool QgsGpkgFlusher::isStopped() const
+{
+  return mFlusher->isStopped();
+}
+
 void Flusher::scheduleFlush( const QString &filename )
 {
+  if ( mIsStopped )
```

Comment:
We still want to be able to schedule flushes. We just don't want them to be executed.

Reply:
What is the point to schedule a flush, if it is never going to get executed? I think it is safer to keep this `if ( mIsStopped ) return` in `Flusher::scheduleFlush`.  What will happen if you call `Flusher::stop`, then cann `Flusher::scheduleFlush`, the timer starts to count down, and in the mean time you call `Flusher::start`? It will execute the flushes that were scheduled while the `mIsStopped` was true, which I think is a bit uncertain and hard to explain.

Calling `Flusher::stop` already clears all the running timers, because they are immediately executed.

Probably I misunderstood your comment above. Anyway, I made another commit that combines your suggestions, while keeping the `scheduleFlush` early return.

Reply:
You are right.

My concern is the situation where you stop for a file A and while it's stopped you want to flush B and loose the event. You'd probably rather want to just defer that to when it's started again.

Reply:
For my usecase this would not be the issue, since the project will be unloaded just after the flush. However, I do agree for the general use it might cause troubles and would be safer to keep the flush for later.
What if we stop the timer by filename `Flusher::stop( const QString &fileName )`? This way we can still have the B flushed and A ignoring flushing requests.

Reply:
That would be very clean indeed.
And I also agree that it's more an aesthetic issue at the moment - and coule also be addressed with proper doc strings at the current point in time.

Reply:
I already gave it a try, this should do the trick.